optimize/upgrade/src/main/java/io/camunda/optimize/upgrade/steps/schema/DeleteIndexTemplateIfExistsStep.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.optimize.upgrade.steps.schema;
+
+import static io.camunda.optimize.upgrade.steps.UpgradeStepType.SCHEMA_DELETE_TEMPLATE;
+
+import io.camunda.optimize.service.db.schema.IndexMappingCreator;
+import io.camunda.optimize.service.db.schema.OptimizeIndexNameService;
+import io.camunda.optimize.upgrade.es.SchemaUpgradeClient;
+import io.camunda.optimize.upgrade.exception.UpgradeRuntimeException;
+import io.camunda.optimize.upgrade.steps.UpgradeStep;
+import io.camunda.optimize.upgrade.steps.UpgradeStepType;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+
+@EqualsAndHashCode(callSuper = true)
+public class DeleteIndexTemplateIfExistsStep extends UpgradeStep {
+
+  // This should be the name of the template prefix and without version suffix or optimize prefix
+  @Getter private final String templateName;
+  @Getter private final int templateVersion;
+
+  public DeleteIndexTemplateIfExistsStep(final String templateName, final int templateVersion) {
```

Comment:
struggle to review some syntax here just because of lack of knowledge. Have you tested it though manually to make sure the template is deleted as expected in the upgrade?

Reply:
I ran the MigrationIT locally, is it possible to run an upgrade manually without a release? Would I just package it to create my own?

Reply:
Lets cover this during QA